operator/pkg/central/extensions/common_test.go

```diff
@@ -165,7 +165,7 @@ func testSecretReconciliation(t *testing.T, runFn func(ctx context.Context, cent
 	assert.Empty(t, secretsByName, "one or more unexpected secrets exist")
 
 	// Verify that a second invocation does not further change the cluster state
-	err = runFn(context.Background(), central.DeepCopy(), client, statusUpdater, logr.Discard())
+	err = runFn(context.Background(), central.DeepCopy(), client, client, statusUpdater, logr.Discard())
```

Comment:
One thing to note is that in almost all unit tests, I'm setting the fake client for both Client and APIReader. This is because the fakeclient is always considered source of truth for cache/API reads.